.teamcity/_self/projects/WPComPlugins.kt

```diff
@@ -61,6 +63,91 @@ object WPComPlugins : Project({
 	}
 })
 
+object BuildPlugins: BuildType({
+	id("calypso_WPComPlugins_Build_Plugins")
+	uuid = "8453b8fe-226f-4e91-b5cc-8bdad15e0814"
+	name = "Build WPCom Plugins"
+	description = "Test"
+
+	vcs {
+		root(Settings.WpCalypso)
+		cleanCheckout = true
+	}
+
+	artifactRules = """
+		apps/happy-blocks/release-files => happy-blocks.zip
+		apps/notifications/dist => notifications.zip
+	""".trimIndent()
+
+	steps {
+		mergeTrunk()
+
+		bashNodeScript {
+			name = "Prepare environment"
+			scriptContent = """
+				set -x
+
+				# Update composer
+				composer install
+
+				# Install dependencies
+				yarn
+
+				# Set execution permission for additional scripts.
+				chmod +x .teamcity/scripts/WPComPlugins/
+			"""
+		}
+
+		bashNodeScript {
+			name = "Build artifacts"
+			scriptContent = """
+				# @automattic/wpcom-block-editor needs this variable.
+				export NODE_ENV="development"
+
+				# Run `yarn build` for the plugins specified in the glob.
+				yarn workspaces foreach --verbose --parallel --include '{happy-blocks,@automattic/notifications}' run build
+
+				# Run extra script necessary for Happy Blocks.
+				# When other plugins that need additional configuration are added,
+				# this should be parallelized.
+				./.teamcity/scripts/WPComPlugins/happy-blocks.sh &
+			"""
+		}
+
+		/**
+		 * We download the archive directly in this step rather than relying on
+		 * an artifact dependency. We do this because if two commits on trunk
+		 * build at the same time, then they will both point to the same artifact
+		 * dependency. In this scenario, we actually want the current build to
+		 * diff against the artifact from that other commit build.
+		 *
+		 * Using the artifact dependency feature, we can only rely on already-finished
+		 * builds at the time the current build *starts*. This means every build
+		 * would have to run in serial, which is not possible in TeamCity without
+		 * a plugin. As a result, commits have to happen several minutes apart
+		 * in order for the diff tagging feature to work correctly in this scenario.
+		 *
+		 * Downloading from the API directly means that the previous build only
+		 * has to finish by the time this *step* begins. As a result, as long as
+		 * the two builds start further apart than the time this step takes,
+		 * then we can diff against the correct artifact. This means two builds
+		 * can be started within a few seconds of each other on trunk, and the
+		 * most recent build of the two can still rely on the other's artifact.
+		 */
+		bashNodeScript {
+			name = "Process Artifacts"
+			scriptContent = """
+				# Plugins that need to have artifacts processed.
+				# Pass in the directory as it appears on disk without the `app/` directory.
+				declare pluginDirs=("happy-blocks" "notifications")
+
+				./.teamcity/scripts/WPComPlugins/processArtifacts.sh \
+					"happy-blocks" "notifications"
```

Comment:
Another thought is that there's no real need for process artifacts to be separate. For example, as soon as the `yarn build` for notifications completes, its artifact could be processed right away

Reply:
@noahtallen I've pushed a revision that contains experimental changes that distribute the act of processing and prepping the build output for CI upload, to the `build-ci` action for each plugin.

This means every plugin will need to have its `package.json` updated to call a new `bin/prepArtifactForCI.sh` script which performs the additional steps.

This should, in theory, make the TeamCity build configuration very simple as all it has to do is call the `yarn build-ci` command in each workspace, then locate & upload the artifacts.

Reply:
👍 

What are you thinking for the initial first steps? Should we create the build config in teamcity so we can start testing it?

Reply:
> Should we create the build config in teamcity so we can start testing it?

Yes, I will retitle this PR to make it clear that it is for a proof of concept, hopefully merge it and start iterating on it.